tests/installation/first_boot.pm

```diff
@@ -52,30 +76,25 @@ sub run() {
             type_password;
             send_key 'ret';
         }
-        handle_login;
     }
-
-    my @tags = qw(generic-desktop);
-    if (check_var('DESKTOP', 'kde') && get_var('VERSION', '') =~ /^1[23]/) {
-        push(@tags, 'kde-greeter');
-    }
-    # GNOME and KDE get into screenlock after 5 minutes without activities.
-    # using multiple check intervals here then we can get the wrong desktop
-    # screenshot at least in case desktop screenshot changed, otherwise we get
-    # the screenlock screenshot.
-    my $timeout        = 600;
-    my $check_interval = 30;
-    while ($timeout > $check_interval) {
-        my $ret = check_screen \@tags, $check_interval;
-        last if $ret;
-        $timeout -= $check_interval;
-    }
-    # the last check after previous intervals must be fatal
-    assert_screen \@tags, $check_interval;
-    if (match_has_tag('kde-greeter')) {
-        send_key "esc";
-        assert_screen 'generic-desktop';
+    else {
+        handle_first_desktop;
+        # hide mouse for clean logout needles
+        mouse_hide();
+        # logout to test login on openSUSE https://progress.opensuse.org/issues/13306
+        if (check_var('DESKTOP', 'gnome') || check_var('DESKTOP', 'lxde')) {
+            my $command = check_var('DESKTOP', 'gnome') ? 'gnome-session-quit' : 'lxsession-logout';
+            x11_start_program("$command");    # opens logout dialog
+            assert_screen 'logoutdialog' unless check_var('DESKTOP', 'gnome');
+        }
+        else {
+            my $key = check_var('DESKTOP', 'xfce') ? 'alt-f4' : 'ctrl-alt-delete';
+            send_key_until_needlematch 'logoutdialog', "$key";    # opens logout dialog
+        }
+        assert_and_click 'logout-button';                         # press logout
     }
+    handle_login;
```

Comment:
That optimistically assumes the login prompt is already there right after the logout-button is clicked?

Especially in GNOME this fails as handle_login goes right to press 'RET" to select the user. But if gdm is not yet on screen/loaded, we fail to login afterwards..

See for example https://openqa.opensuse.org/tests/333191#step/first_boot/9 => it already expects the login prompt - but apparently the user selection was never done